worlds/marioland2/__init__.py

```diff
@@ -0,0 +1,576 @@
+                item_counts["Cancel Auto Scroll"] = 1
+            else:
+                for level, i in enumerate(self.auto_scroll_levels):
+                    if i == 3:
+                        item_counts[f"Auto Scroll - {level_id_to_name[level]}"] = 1
+                    elif i == 2:
+                        item_counts[f"Cancel Auto Scroll - {level_id_to_name[level]}"] = 1
+
+        for item in self.multiworld.precollected_items[self.player]:
+            if item.name in item_counts and item_counts[item.name] > 0:
+                item_counts[item.name] -= 1
+
+        location_count = len(self.multiworld.get_unfilled_locations(self.player))
+        items_to_add = location_count - sum(item_counts.values())
+        if items_to_add > 0:
+            mario_coin_frags = 0
+            if self.options.shuffle_golden_coins == "mario_coin_fragment_hunt":
+                mario_coin_frags = (items_to_add * self.options.mario_coin_fragment_percentage) // 100
+                item_counts["Mario Coin Fragment"] += mario_coin_frags
+            item_counts["Super Star Duration Increase"] += items_to_add - mario_coin_frags
+        elif items_to_add < 0:
+            if self.options.coinsanity:
+                for i in range(1, 168):
+                    coin_name = f"{i} Coin{'s' if i > 1 else ''}"
+                    if coin_name in item_counts:
+                        amount_to_remove = min(-items_to_add, item_counts[coin_name])
+                        item_counts[coin_name] -= amount_to_remove
+                        items_to_add += amount_to_remove
+                        if items_to_add >= 0:
+                            break
+
+            double_progression_items = ["Tree Zone Progression", "Macro Zone Progression", "Pumpkin Zone Progression",
+                                        "Mario Zone Progression", "Turtle Zone Progression"]
+            self.random.shuffle(double_progression_items)
+            while sum(item_counts.values()) > location_count:
+                if double_progression_items:
+                    double_progression_item = double_progression_items.pop()
+                    item_counts[double_progression_item] -= 2
+                    item_counts[double_progression_item + " x2"] = 1
+                    continue
+                if self.options.auto_scroll_mode in ("level_trap_items", "level_cancel_items",
+                                                     "chaos"):
+                    auto_scroll_item = self.random.choice([item for item in item_counts if "Auto Scroll" in item])
+                    level = auto_scroll_item.split("- ")[1]
+                    self.auto_scroll_levels[level_name_to_id[level]] = 0
+                    del item_counts[auto_scroll_item]
+                    continue
+                raise Exception(f"Too many items in the item pool for Super Mario Land 2 player {self.multiworld.player_name[self.player]}")
```

Comment:
You can use this instead
```suggestion
                raise Exception(f"Too many items in the item pool for Super Mario Land 2 player {self.player_name}")
```